pkg/oc/cli/newapp/newapp_test.go

```diff
@@ -381,21 +385,21 @@ func TestNewAppRunQueryActions(t *testing.T) {
 
 	for _, test := range tests {
 		// Prepare structure for test.
-		templateClient := templatefake.NewSimpleClientset(fakeTemplateList())
-		imageClient := imagefake.NewSimpleClientset(fakeImagestreamList())
+		templateClient := templatev1fake.FakeTemplateV1{Fake: &(templatefake.NewSimpleClientset(fakeTemplateList()...).Fake)}
+		imageClient := imagev1fake.FakeImageV1{Fake: &(imagefake.NewSimpleClientset(fakeImagestreamList()).Fake)}
 
 		o.Config = test.config
 		o.Config.Deploy = true
 
-		o.Config.SetOpenShiftClient(imageClient.Image(), templateClient.Template(), nil, "openshift", nil)
+		o.Config.SetOpenShiftClient(&imageClient, &templateClient, nil, "openshift", nil)
 
 		var dockerVisited, tfVisited bool
 		o.Config.DockerSearcher = MockSearcher{
 			OnSearch: func(precise bool, terms ...string) (app.ComponentMatches, []error) {
 				dockerVisited = true
 				match := &app.ComponentMatch{
-					Name:  "repo/test",
-					Image: &imageapi.DockerImage{},
+					Name:        "repo/test",
+					DockerImage: &imageapi.DockerImage{},
```

Comment:
Use the `docker10` import from openshift's API like you did in `ComponentMatch` struct.